beets/dbcore/db.py

```diff
@@ -201,21 +234,30 @@ def get(self, key, default=None):
         else:
             return default
 
-    def __contains__(self, key) -> bool:
+    def __contains__(self, key: Any) -> bool:
         """Determine whether `key` is an attribute on this object.
         """
-        return key in self.keys()
+        return key in self._converted or key in self._data
 
-    def __iter__(self) -> Iterable[str]:
+    def __iter__(self) -> Iterator[str]:
         """Iterate over the available field names (excluding computed
         fields).
         """
+        # NOTE: It would be nice to use the following:
+        # yield from self._converted
+        # yield from self._data
+        # but that won't work since some code relies on modifying `self`
+        # during iteration.
         return iter(self.keys())
 
+    def __len__(self) -> int:
+        # FIXME: This is incorrect due to duplication of keys
+        return len(self._converted) + len(self._data)
```

Comment:
Would it work to do `len(self.keys())`, even if that's obviously inefficient? I suppose this probably doesn't matter much!

Reply:
Actually no, because one of the peculiarities of `LazyConvertDict` is that `keys` gives a list, which as far as I understand, should have the same duplicates.

I have some code locally that changes the implementation (basically, `pop` entries from `_data` when adding to `_converted`) to avoid this.

But maybe I'm just misunderstanding something here: It is somewhat surprising that no issues were caused by such duplicates (but maybe, we just don't have much code that iterates `LazyConvertDict`?).

Reply:
Ah, got it! So this essentially reflects a regrettable quirk, but it reflects that quirk accurately.

Yes, I agree with you that this hasn't caused problems historically. I agree that there probably are not too many places where we iterate over the mapping, and where we do, the situation could easily be idempotent (e.g., when inserting into another map, which will deduplicate things). 🤷